### PR TITLE
[NEW] Send message with attachments

### LINKED ISSUE #944: [NEW] Allow sending message with attachments

### PR DESCRIPTION
**Pull request by divyanshub024**
@RocketChat/android

Closes #944 
Merge #1249 first

- Image

- Video

- File

### CODE REVIEW

**Comment by philipbrito** [comment]
@divyanshub024 I loved seeing the puppy 🐶. :blush:

**Comment by divyanshub024** [comment]
@filipedelimabrito Haha.. Thanks! It took a while to find a nice pic :smile:

**Comment by divyanshub024** [comment]
@filipedelimabrito ci/circleci: code-analysis is failing. Should I make changes or leave it?
```  The first 3 errors (out of 8) were:
  /home/circleci/project/app/src/main/res/menu/chatroom_actions.xml:8: Error: Should use android:showAsAction when not using the appcompat library [AppCompatResource]
          app:showAsAction="never" />
          ~~~~~~~~~~~~~~~~~~~~~~~~
  /home/circleci/project/app/src/main/res/menu/chatroom_actions.xml:13: Error: Should use android:showAsAction when not using the appcompat library [AppCompatResource]
          app:showAsAction="never" />
          ~~~~~~~~~~~~~~~~~~~~~~~~
  /home/circleci/project/app/src/main/res/menu/chatroom_actions.xml:18: Error: Should use android:showAsAction when not using the appcompat library [AppCompatResource]
          app:showAsAction="never" />
          ~~~~~~~~~~~~~~~~~~~~~~~~```

**Comment by philipbrito** [comment]
It will be good if you can fix here. But if you have no time we can look at it later 👍

**Comment by divyanshub024** [comment]
@filipedelimabrito I'll try to fix it tomorrow in a new PR.

**Comment by philipbrito** [comment]
@divyanshub024 Could you fix the conflicts?

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 1]
```diff
@@ -23,6 +24,11 @@ import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.DefaultItemAnimator
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
+import android.text.SpannableStringBuilder
+import android.view.*
```
Comment (luciofm) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Don't use wildcard imports please.

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt [hunk 2]
```diff
@@ -216,9 +224,51 @@ class ChatRoomFragment : Fragment(), ChatRoomView, EmojiKeyboardListener, EmojiR
...
+    }
+
+    private fun fileAttachmentDialog(data: Uri) {
+        val builder = AlertDialog.Builder(activity)
+        val dialogView = View.inflate(context, R.layout.file_attachments_dialog, null)
+        builder.setView(dialogView)
+        val alertDialog = builder.create()
+
+        dialogView?.let {
+            val imagePreview: ImageView = it.findViewById(R.id.image_preview)
```
Comment (luciofm) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Don't need to change, but a more Kotlin idiomatic syntax would be:
`val imagePreview = it.findViewById<ImageView>(R.id.image_preview)`

**Review by luciofm — CHANGES_REQUESTED** [review]

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt (hunk 1 above)
Comment (divyanshub024) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: Sure! But can you tell me why it's the bad idea.

**Comment by divyanshub024** [comment]
@filipedelimabrito @luciofm Requested changes have been made

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt (hunk 1 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt]: There are many recommendations about why using wildcard is bad (read this [article](http://www.javadude.com/articles/importondemandisevil.html) for instance), in the other side some people advocates by using wildcard (there is even a recommendation for doing that in the Clean Code book).

In short, by not using the wildcard won't fail the build when we add the lint task here in the future (like we have for our Kotlin SDK).

**Comment by philipbrito** [comment]
Thanks @divyanshub024 !
Unfortunately this PR got new conflicts after merging some other PR's. Could you fix them one more time?

Note: `app/src/main/res/menu/chatroom_actions.xml` is gone now.

**Comment by divyanshub024** [comment]
@filipedelimabrito I have resolved the conflicts.

**Comment by divyanshub024** [comment]
@filipedelimabrito This PR is ready.

**Review by philipbrito — APPROVED** [review]
Thank you very much for this PR @divyanshub024 !!!
💯

**Pull request merged**